src/pages/_app.tsx

```diff
@@ -3,6 +3,15 @@ import SiteLayout from '../components/global/SiteLayout'
 import { Web3ReactProvider, createWeb3ReactRoot } from '@web3-react/core'
 import { ethers } from 'ethers'
 import { DefaultProviderName } from 'src/constants'
+import { isProduction } from 'src/utils'
+
+// yarn dev issue seems due to web3react provider trying to run on the server due to next's SSR
+// typeof window !== 'undefined' checks if code is on the browser or server
+// ternary checks to see if on browser or server and returns null if on server
+// TODO instead of disabling SSR locally, create a flow that serves a basic static site
+// and hydrates Web3 client side; otherwise we get stuck with the hydration issue `Warning: Expected server HTML to contain a matching...`
+const useSSR = isProduction() || typeof window !== 'undefined'
```

Comment:
Let's use dynamic components for this. [dynamic-import#with-no-ssr](https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr)
It provides a solution for exactly this problem, without needing to know environment variables.

Reply:
Would you mind putting that together?

Reply:
Yes, I will do it.

Reply:
Looks like Tgriffith beat you to it https://github.com/BanklessDAO/bankless-dao-website/pull/48

Reply:
that's actually perfect! too busy these days